packages/neos-ui/src/Containers/PrimaryToolbar/DimensionSwitcher/DimensionSelector.js

```diff
@@ -0,0 +1,112 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import Icon from '@neos-project/react-ui-components/src/Icon/';
+import SelectBox from '@neos-project/react-ui-components/src/SelectBox/';
+import style from './style.module.css';
+import {$get, $transform} from 'plow-js';
+import mapValues from 'lodash.mapvalues';
+import I18n from '@neos-project/neos-ui-i18n';
+import sortBy from 'lodash.sortby';
+import {neos} from '@neos-project/neos-ui-decorators';
+import DimensionSelectorOption from './DimensionSelectorOption';
+
+const searchOptions = (searchTerm, processedSelectBoxOptions) =>
+    processedSelectBoxOptions.filter(option => option.label && option.label.toLowerCase().indexOf(searchTerm.toLowerCase()) !== -1);
+
+@neos(globalRegistry => ({
+    i18nRegistry: globalRegistry.get('i18n')
+}))
+export default class DimensionSelector extends PureComponent {
+    static propTypes = {
+        icon: PropTypes.string.isRequired,
+        dimensionLabel: PropTypes.string.isRequired,
+        presets: PropTypes.object.isRequired,
+        activePreset: PropTypes.string.isRequired,
+        dimensionName: PropTypes.string.isRequired,
+        isLoading: PropTypes.bool,
+        onSelect: PropTypes.func.isRequired,
+        showOnlyDimensionDropdown: PropTypes.bool,
```

Comment:
The name of the prop doesn't really tell much about what is the alternative. 

Wouldn't it be better to leave the list item wrapping out of this component and decide the wrapping in the DimensionSwitcher?

Reply:
Changed that now. Thanks for the feedback.